cmd/kar-controllers/app/server.go

```diff
@@ -62,6 +64,12 @@ func Run(opt *options.ServerOption) error {
 		AgentConfigs: strings.Split(opt.AgentConfigs, ","),
 	}
 
+	if err = discovery.InitializeGlobalDiscoveryClient(restConfig); err != nil {
+		klog.Errorf("Error initializing global discovery client: %s", err)
+	} else {
+		klog.Infof("Initializing global discovery client")
```

Comment:
I think adding the log level here makes sense as this is informational, so can be filtered out at other log levels. 
```suggestion
		klog.V(4).Infof("Initializing global discovery client")
```

Reply:
Hey Fiona, 
Thanks for the review. I'll update the log :)

The refresh timeframe set to 5 hours is an arbitrary number, I am aiming for a timeframe which both promotes the efficiency of the cache, whist not holding cache for too long, therefor I chose to half that of the controller-runtime resync period of 10 hours. 

Although I am open for discussion if you have any suggestions towards the timeframe :)

Reply:
Just thinking about it, and how it would work. From what I understand the client only initializes once, and then refreshes after 5 hours. If resources are created soon after a refresh, will they then only be available for manipulation after the next refresh?

Reply:
If you query for an object type which doesn't exist in the cache it would cause a cache miss and invalidation triggering queries to the k8 discovery client to repopulate the cache.

Reply:
Thanks Kevin. So if a new object of an object type that does exist is created, this will be discoverable?

Reply:
Yes, but the discovery client cache doesn't cache individual k8 objects. As I understand it, the Discovery Client is the piece of k8 which resolves names such as `pod` or `po` into a fully resolved kubernetes API endpoint. Without a cache, every time one tries to get a pod from kubernetes it will iterate through the discovery API until it finds an entry with that name. This means that in the worst case that every resource will be checked using a `GET` request which can cause client throttling. Without a cache this would happen every time.

Reply:
Thanks for explaining!